src/utils.jl

```diff
@@ -69,11 +69,10 @@ function invwsumsq(w::AbstractVector, a::AbstractVector)
 end
 
 function colwise_dot!(r::AbstractArray, a::AbstractMatrix, b::AbstractMatrix)
-    n = length(r)
-    @check_argdims n == size(a, 2) == size(b, 2) && size(a, 1) == size(b, 1)
-    for j = 1:n
+    @check_argdims size(a, 2) == size(b, 2) && size(a, 1) == size(b, 1)
```

Comment:
They have to be using the same axes.
As we index them with the same indicies
```suggestion
    @check_argdims(axes(a) == axes(b))
```